File: producer/tests/mock_updates_provider.py
from datetime import datetime

from producer.kafka.iupdates_provider import IUpdatesProvider
from producer.updateapi.update import Update


class MockUpdatesProvider(IUpdatesProvider):
    def get_updates(self, user_id: str):
        return [
            Update(
                content='Mock updateapi',
                author_id='Mock author',
                creation_date=datetime.now(),
                url='mockurl://updateapi.com',
                media=[])
        ]

File: UpdatesScraper.Tests/MockUpdatesProvider.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Common;

namespace UpdatesScraper.Tests
{
    internal class MockUpdatesProvider : IUpdatesProvider
    {
        public Task<IEnumerable<Update>> GetUpdatesAsync(User user)
        {
            IEnumerable<Update> updates = new Update[]
            {
                new()
                {
                    Content = $"mock update by {user}",
                    Author = user,
                    CreationDate = DateTime.Now,
                    Url = "mockurl.com"
                }
            };
            
            return Task.FromResult(updates);
        }
    }
}